ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -473,49 +626,62 @@ def set_waypoints_velocity(self):
-                      .format(dist_to_tl, stopping_distance, self.state))
+            if (math.fabs(self.prev_step_v -
+                    self.waypoints[self.final_waypoints_start_ptr].get_v()) > 0.1
+                    or 
+                    math.fabs(self.prev_step_a -
+                    self.waypoints[self.final_waypoints_start_ptr].get_a()) > 0.1):
+                self.stopping_distance = get_accel_distance(
+                    self.waypoints[self.final_waypoints_start_ptr].get_v(),
+                    0.0, -self.default_accel/accel_ratio,
+                    self.waypoints[self.final_waypoints_start_ptr].get_a())
+                self.min_stop_distance = self.get_min_stopping_distance(
+                    self.final_waypoints_start_ptr)
+
+        rospy.logdebug("dist_to_tl = {:4.3f}, stopping_dist = {:4.3f}, state = "
+                      "{} min_stopping_distance = {}"
+                      .format(dist_to_tl, self.stopping_distance, self.state,
+                              self.min_stop_distance))
 
         # handle case where car is stopped at lights and light is red
         if self.state == 'stopped' and dist_to_tl < self.dyn_tl_buffer:
             self.set_stopped(self.final_waypoints_start_ptr,
                              self.lookahead_wps)
 
-        elif dist_to_tl < stopping_distance + self.dyn_tl_buffer or\
+        elif dist_to_tl < self.stopping_distance + self.dyn_tl_buffer or\
                 dist_to_tl < 20:    # added last check in to try to prevent
                                     # flipping between accel and decel
                                     # when approaching the light
 
             if dist_to_tl < self.dyn_tl_buffer:
                 # small buffer from stop line - stop the car if in this area
-                rospy.loginfo("Now within tl_buffer = {:4.3f}"
+                # Todo evaluate if we need to keep going
+                rospy.logwarn("Now within tl_buffer = {:4.3f} - set stopped"
                               .format(self.dyn_tl_buffer))
                 self.set_stopped(self.final_waypoints_start_ptr, self.
                                  lookahead_wps)
             else:
                 # reduce car velocity to 0.0 stoping before lights
                 if self.state != 'slowdown':
-                    rospy.loginfo("now start to slowdown")
+                    rospy.logwarn("now start to slowdown")
                     self.state = 'slowdown'
-                recalc = self.produce_slowdown(self.final_waypoints_start_ptr,
+                if dist_to_tl - self.dyn_tl_buffer > self.min_stop_distance:
+                    recalc = self.produce_slowdown(self.final_waypoints_start_ptr,
                                                self.lookahead_wps,
-                                               accel_ratio)
+                                               dist_to_tl - self.dyn_tl_buffer)
```

Comment:
If the dist_to_tl activates within the normal (self.stopping_distance + buffer) and is also already closer than the self.min_stop_distance, this means that the car can't stop in time and will just keep going?  If so, can you add an else condition here that at least makes a log notification of this so we can confirm this behavior during testing?